script/changelog/api.ts

```diff
@@ -1,60 +1,128 @@
-      },
-    }
+interface IAPIPullRequest {
+  readonly title: string
+  readonly body: string
+  readonly user: IAPIUser
+}
 
-    const request = HTTPS.request(options, response => {
-      let received = ''
-      response.on('data', chunk => {
-        received += chunk
-      })
-
-      response.on('end', () => {
-        try {
-          const json: GraphQLResponse = JSON.parse(received)
-          const pr = json.data.repository.pullRequest
-          resolve(pr)
-        } catch (e) {
-          resolve(null)
-        }
-      })
+interface IAPICommit {
+  readonly sha: string
+  readonly author?: IAPIUser
+  readonly committer?: IAPIUser
+}
+
+interface IAPITeam {
+  readonly name: string
+  readonly id: number
+}
+
+interface IAPITeamMember {
+  readonly login: string
+}
+
+// this account is assigned by GitHub as the committer for merged pull requests
+// and should be excluded from being considered an external contributor
+const webflowAccount = ['web-flow']
+
+export async function getCoreTeamMembers(): Promise<ReadonlyArray<string>> {
+  try {
+    let response = await octokit.orgs.getTeams({
+      org: 'desktop',
+      per_page: 100,
     })
+    const teams: ReadonlyArray<IAPITeam> = response.data
+    const coreTeam = teams.find(t => t.name === 'Core') || null
 
-    const graphql = `
-{
-  repository(owner: "desktop", name: "desktop") {
-    pullRequest(number: ${id}) {
-      title
-      body
+    if (coreTeam == null) {
+      console.error('Unable to find core team on API')
+      return []
     }
+
+    const id = coreTeam.id
+
+    response = await octokit.orgs.getTeamMembers({
+      id,
+      role: 'all',
+      per_page: 100,
+    })
+    const members: ReadonlyArray<IAPITeamMember> = response.data
+
+    return members.map(m => m.login).concat(webflowAccount)
+  } catch (err) {
+    console.error('API lookup failed for getCoreTeamMembers', err)
+    return []
   }
 }
-`
-    request.write(JSON.stringify({ query: graphql }))
 
-    request.end()
-  })
+export async function fetchPR(id: number): Promise<IDesktopPullRequest | null> {
+  try {
+    const pullRequestResponse = await octokit.pullRequests.get({
+      owner: 'desktop',
+      repo: 'desktop',
+      number: id,
+    })
+    const { title, body } = pullRequestResponse.data as IAPIPullRequest
+
+    let commitsResponse = await octokit.pullRequests.getCommits({
+      owner: 'desktop',
+      repo: 'desktop',
+      number: id,
+      per_page: 100,
+    })
+
+    let data: Array<IAPICommit> = commitsResponse.data
+    while (octokit.hasNextPage(commitsResponse)) {
+      commitsResponse = await octokit.getNextPage(commitsResponse)
+      data = data.concat(commitsResponse.data)
+    }
+
+    const collaborators: Array<string> = []
+    const commits: Array<string> = []
+
+    for (const commit of data) {
+      commits.push(commit.sha)
+      if (commit.author && collaborators.indexOf(commit.author.login) === -1) {
```

Comment:
this took some squinting so i tried moving things around so their easier to read; i pulled out consts in the process so if space is a concern or there doesn't feel like there is much to gain, feel free to ignore:
```
    for (const commit of data) {
      const { sha, author, committer } = commit
      commits.push(sha)
      if (author != null && !(author.login in collaborators)) {
        collaborators.push(author.login)
      }
      if (committer != null && !(committer.login in collaborators)) {
        collaborators.push(committer.login)
      }
    }
```

Reply:
at minimum we can make the first check in `if (commit.author &&` and the one below more explicit, ex: commit.author != null

Reply:
I went one better than your snippet and used a `Set<string>` here:

```
if (author != null && !collaborators.has(author.login)) {
  collaborators.add(author.login)
}
```